src/transformers/processing_utils.py

```diff
@@ -1261,17 +1325,51 @@ def apply_chat_template(
                             video = [np.array(load_image(image_fname)).T for image_fname in fname]
                             # create a 4D video because `load_video` always returns a 4D array
                             video = np.stack(video)
+                            metadata = None
+                            logger.warning(
+                                "When loading the video from list of images, we cannot infer metadata such as `fps` or `duration`. "
+                                "If you model applies special processing based on metadata, please load the whole video and let the model sample frames."
+                            )
                         else:
-                            video = load_video(fname, num_frames=num_frames, fps=video_fps, backend=video_load_backend)
+                            video, metadata = load_video(
+                                fname,
+                                num_frames=num_frames,
+                                fps=video_fps,
+                                backend=video_load_backend,
+                                sample_indices_fn=sample_indices_fn,
+                            )
                         videos.append(video)
+                        video_metadata.append(metadata)
 
-                # Currently all processors can accept accept nested list of batches, but not flat list of visuals
+                # Currently all processors can accept nested list of batches, but not flat list of visuals
                 # So we'll make a batched list of images and let the processor handle it
                 if images:
                     batch_images.append(images)
                 if videos:
                     batch_videos.append(videos)
+                    batch_video_metadata.append(video_metadata)
+
+            # Process conversation with video/image information if needed. Then convert into a prompt using Jinja template
+            conversations = self._process_messaged_for_chat_template(
```

Comment:
to change in other locations
```suggestion
            conversations = self._process_messages_for_chat_template(
```